cli/waiter/subcommands/ping.py

```diff
@@ -1,61 +1,88 @@
         if resp.status_code == 200:
-            print(terminal.success(f'Successfully pinged {token_name} in {cluster_name}.'))
+            print(terminal.success(f'Ping successful.'))
             print(resp.text)
             return True
         else:
             print_error(response_message(resp.json()))
             return False
     except Exception:
-        message = f'Encountered error while pinging token {token_name} in {cluster_name}.'
+        message = f'Encountered error while pinging in {cluster_name}.'
         logging.exception(message)
         print_error(message)
 
 
+def ping_token_on_cluster(cluster, token_name, health_check_endpoint, timeout):
+    """Pings the token with the given token name in the given cluster."""
+    cluster_name = cluster['name']
+    print(f'Pinging token {terminal.bold(token_name)} '
+          f'at {terminal.bold(health_check_endpoint)} '
+          f'in {terminal.bold(cluster_name)}...')
+    return ping_on_cluster(cluster, health_check_endpoint, timeout, token_name)
+
+
+def ping_service_on_cluster(cluster, service_id, health_check_endpoint, timeout):
+    """Pings the service with the given service id in the given cluster."""
+    cluster_name = cluster['name']
+    print(f'Pinging service {terminal.bold(service_id)} '
+          f'at {terminal.bold(health_check_endpoint)} '
+          f'in {terminal.bold(cluster_name)}...')
+    return ping_on_cluster(cluster, health_check_endpoint, timeout, f'^SERVICE-ID#{service_id}')
+
+
 def ping(clusters, args, _):
     """Pings the token with the given token name."""
     guard_no_cluster(clusters)
-    token_name = args.get('token')[0]
-    query_result = query_token(clusters, token_name)
+    token_name_or_service_id = args.get('token-or-service-id')
+    is_service_id = args.get('is-service-id', False)
+    if is_service_id:
+        query_result = query_service(clusters, token_name_or_service_id)
+    else:
+        query_result = query_token(clusters, token_name_or_service_id)
+
     if query_result['count'] == 0:
         print_no_data(clusters)
         return 1
 
+    timeout = args.get('timeout', None)
     http_util.set_retries(0)
     cluster_data_pairs = sorted(query_result['clusters'].items())
     clusters_by_name = {c['name']: c for c in clusters}
     overall_success = True
     for cluster_name, data in cluster_data_pairs:
         cluster = clusters_by_name[cluster_name]
-        health_check_endpoint = data['token'].get('health-check-url', '/status')
-        success = ping_token_on_cluster(cluster, token_name, health_check_endpoint, args.get('timeout', None))
+        if is_service_id:
+            health_check_endpoint = data['service']['effective-parameters'].get('health-check-url', '/status')
+            success = ping_service_on_cluster(cluster, token_name_or_service_id, health_check_endpoint, timeout)
+        else:
+            health_check_endpoint = data['token'].get('health-check-url', '/status')
```

Comment:
Should we retrieve the default, here and above, from `/settings` endpoint?

Reply:
Discussed offline. I removed the default from the service id path because we should always get it back when using `effective-parameters`. For the token path, I added a comment about the fact that we're hard-coding it.